projects/machine_learning.md

```diff
@@ -0,0 +1,71 @@
+4. **Train and validate the model**
+Split the data into a train and test set. Split the train set again if you need to optimise the hyper-parameters. Make sure to make a good choice for the splits, some types of data require specialised splits (e.g. time series). Train the model and assess its predictive power on the test data set. Is the training successful? If the training procedure is iterative: how did you determine the endpoint?
+5. **Interpret the model**
+The concept of model interpretability has now [reached the European parliament](https://www.europarl.europa.eu/news/en/headlines/society/20230601STO93804/eu-ai-act-first-regulation-on-artificial-intelligence). Try to understand your model: how does it behave? What are the most important input features? Is the full input space captured well? Do you see any trends in residuals?
+
+Suggested extensions:
+
+- Some data can benefit from feature engineering: applying transformations on your data to incorporate some expert knowledge. As an example: it might be that only the magnitude of a feature is important, so one might use the absolute value of the feature.
+
+## 4. Resources
+
+For this project, we heavily encourage you to use the [MLJ](https://alan-turing-institute.github.io/MLJ.jl/stable/#Data) package. Second, we also encourage the use of DrWatson.jl to forge some good habits regarding code structure and reproducibility of data science projects.
+
+1. In case that you do not want to use your own data set, we recommend one of the following Julia packages:
+- [MLDatasets.jl](https://juliaml.github.io/MLDatasets.jl/stable/)
+- [ForecastData.jl](https://github.com/viraltux/ForecastData.jl)
+- [MarketData.jl](https://juliaquant.github.io/MarketData.jl/stable/company_financial_series/#Large-historical-data-sets-1)
+- [RDatasets.jl](https://github.com/JuliaStats/RDatasets.jl)
+- [CDSAPI.jl](https://github.com/JuliaClimate/CDSAPI.jl)
+- [HuggingFaceDatasets.jl](https://github.com/CarloLucibello/HuggingFaceDatasets.jl)
+    
+    Another option is to go the synthetic route. This might be interesting if you already have a specific model in mind that you want to use and you want to assess the effect of the dimensionality and noise on your performance. 
+    
+- [Synthetic data from MLJ](https://alan-turing-institute.github.io/MLJ.jl/stable/generating_synthetic_data/)
+2. For visualisation you can use the Plots.jl package (make your own Plotrecipe if you are feeling fancy), consider if some dimensionality reduction or manifold learning would be interesting for your chosen data set
+3. Hint:
+
+```julia
+X = ...
+y = ...
+using MLJ
+models(matching(X, y))
+```
+
+4. If you are keen to try out neural networks: we recommend using the [Flux](https://fluxml.ai/) ecosystem of packages. 
+5. Consider using pipelines and target transformers for reproducible results!
```

Comment:
can you add a url that gives some examples/documentation to `pipelines` and `target transformers`.